package.json

```diff
@@ -3,15 +3,15 @@
   "version": "0.0.1",
   "scripts": {
     "ng": "ng",
-    "start": "ng serve",
+    "start": "concurrently \"npm run start:api\" \"ng serve\"",
     "start:api": "cd api && npm start",
     "build": "ng build",
     "watch": "ng build --watch --configuration development",
     "test": "jest",
     "test:coverage": "jest --coverage",
-    "test:e2e": "cd e2e && npm run cypress",
-    "test:e2e:local:ci": "cd e2e && npm run cy:local:ci",
-    "test:e2e:ci": "cd e2e && npm run cy:ci"
+    "test:e2e:open": "concurrently \"npm start\" \"cd e2e && npm run cypress\"",
+    "test:e2e:local": "start-server-and-test start http://localhost:4200 \"cd e2e && npm run cy:local\"",
+    "test:e2e:ci": "start-server-and-test start http://localhost:4200 test:e2e:ci"
```

Comment:
You haven't tested this command. This is the command that will run in a CI pipeline, and you're calling recursively the same command in an infinite loop. So, this means that you will block the pipeline indefinitely.

You need to pay more attention to this kind of things, because the consequences of breaking a pipeline are expensive.

To fix this, you need to create the following command in your Cypress package.json: 

```
    "cypress:run": "cypress run"
```

And after that, update this command to:

```
    "test:e2e:ci": "start-server-and-test start http://localhost:4200 \"cd e2e && npm run cypress:run\""
```

Reply:
🤦🏼 you are right, I changed last entry and I didn't test it, now should be solved all the problems

Reply:
Yep, that should work now, good job there! You're starting the front-end app and the back-end app concurrently. If you want, you can avoid starting the back-end for your tests, as you're mocking the responses.

Reply:
I'm not running back-end when I'm running unit test, I'm running when I'm running E2E :)